packages/graphql-playground-middleware-hapi/src/index.ts

```diff
@@ -9,45 +9,40 @@ import {
 const pkg = require('../package.json')
 
 export interface Register {
-  (server: Server, options: MiddlewareOptions, next: any): void
-  attributes?: any
+  (server: Server, options: MiddlewareOptions): void
 }
 
-// tslint:disable-next-line only-arrow-functions
-const hapi: Register = function(server, options: any, next) {
-  if (arguments.length !== 3) {
-    throw new Error(
-      `Playground middleware expects exactly 3 arguments, got ${
-        arguments.length
-      }`,
-    )
-  }
-
-  const { path, route: config = {}, ...rest } = options
-
-  const middlewareOptions: RenderPageOptions = {
-    ...rest,
-    version: pkg.version,
-  }
+export interface Plugin {
+  pkg?: any
+  register: Register
+}
 
-  server.route({
-    method: 'GET',
-    path,
-    config,
-    handler: (request, reply) => {
-      reply(renderPlaygroundPage(middlewareOptions)).header(
-        'Content-Type',
-        'text/html',
+// tslint:disable-next-line only-arrow-functions
+const plugin: Plugin = {
+  pkg,
+  register: function (server, options: any) {
+    if (arguments.length !== 2) {
+      throw new Error(
+        `Playground middleware expects exactly 2 arguments, got ${
+          arguments.length
+        }`,
       )
-    },
-  })
+    }
 
-  return next()
-}
+    const { path, route: config = {}, ...rest } = options
 
-hapi.attributes = {
-  pkg,
-  multiple: false,
+    const middlewareOptions: RenderPageOptions = {
+      ...rest,
+      version: '1.3.6' || pkg.version,
+    }
+
+    server.route({
+      method: 'GET',
+      path,
+      config,
+      handler: async (request, h) => h.response(await renderPlaygroundPage(middlewareOptions)).type('text/html')
```

Comment:
With the latest `graphql-playground-html` there is no need for async await anymore